### ISSUE TITLE
Is this optimization really needed on password reset?

### ISSUE THREAD

**Issue by enricostano** [coopdevs/timeoverflow#327]
I still don't understand what problem we were trying to solve.

Give a look at https://github.com/coopdevs/timeoverflow/commit/f8f940050ac5677805beca4065f9cdf0f7f287bb

**Comment by sauloperez** [comment]
See my comment https://github.com/coopdevs/timeoverflow/commit/f8f940050ac5677805beca4065f9cdf0f7f287bb

**Comment by juanjochi** [comment]
From what I understand it seems that you have to check that when asking for password recovery, you receive an email and you can change it correctly. I have requested the recovery three times, with Opera what I have detected is that the first took several minutes, the other two times the mail arrived in seconds. On the other hand I think that to facilitate the use could hide the button to send the mail again and the unlock and appear after pressing once the password reset button. I have tried from Firefox and it takes seconds to get the recovery mail.

**Comment by enricostano** [comment]
Maybe related https://github.com/coopdevs/timeoverflow/blob/develop/config/initializers/devise.rb#L187

**Comment by markets** [comment]
What are you suggesting @enricostano to increase or decrease that number? Would be amazing to fix this (and remove `memcached` :stuck_out_tongue:) by just tweaking Devise settings.

Another possible approach: to sort out this by db, with something like:

```ruby
def set_reset_password_token
    return reset_password_token if reset_password_sent_at < RESET_PASSWORD_TOKEN_LAZINESS.ago
  end
```

Instead of the current:

https://github.com/coopdevs/timeoverflow/blob/e6ef11997badfcf07a714bb2a44b70757741ddf7/app/models/concerns/lazy_recoverable.rb#L11-L17

**Comment by enricostano** [comment]
I would like to try out Devise settings tuning... and remove the cache.

**Comment by markets** [comment]
@sseerrggii can we just delete this? I'd take care of the ✂️ but I want to confirm first ...

This was added in the past because people are getting different emails (multiple clicks) and only the last one is valid (to reset the pass). But nowadays most of emails clients have good UI for threads/groups, so maybe we can just 🔥 this optimization (=> and remove the entire cache gem and infra piece). 

We can also tune this setting if you want:

```
# Time interval you can reset your password with a reset password key.
# Don't put a too small interval or your users won't have the time to
# change their passwords.
config.reset_password_within = 6.hours
```

If we receive any kind of negative feedback, we can eventually solve it in a different way, like adding proper messages in the UI, or something, but would be amazing to ✂️ this :)

**Comment by markets** [comment]
Just in case @sseerrggii confirms 👉 #517 ✂️ ✂️ ✂️

**Comment by sseerrggii** [comment]
I'm agree, we can remove this optimization! I should close this issue @markets ?

**Comment by markets** [comment]
:):):):) 👍👍👍 

Not necessary to close it now, it will be automatically closed by #517.
